### PR TITLE
FUSETOOLS-1898 - fixing issue with red background on integer parms

### PR DESCRIPTION
**Pull request by bfitzpat**
Not sure why this has taken the whole file and not just added the few lines that were changed.

### CODE REVIEW

**Comment by apupier** [comment]
it should be due to different line ending

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -28,5 +28,6 @@ Require-Bundle: org.junit;bundle-version="4.12.0",
  org.fusesource.ide.camel.model.service.core.tests.integration,
  org.fusesource.ide.branding;bundle-version="9.1.0",
  org.fusesource.ide.projecttemplates;bundle-version="9.1.0"
+Import-Package: org.eclipse.draw2d
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF]: prefer to use require-bundle to stay coherent with other places

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/properties/creators/NumberParameterPropertyUICreatorIT.java [hunk 2]
```diff
@@ -1,76 +1,88 @@
...
+		numberParameterPropertyUICreator.create();
+
+		final Text control = numberParameterPropertyUICreator.getControl();
+		control.setText("2");
+
+		assertThat(modelMap.get("testParameterName")).isEqualTo("2");
+		assertThat(camelModelElement.getParameter("testParameterName")).isEqualTo("2");
+	}
+	
+    @Test
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/properties/creators/NumberParameterPropertyUICreatorIT.java]: wrong indentation

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/properties/creators/NumberParameterPropertyUICreatorIT.java [hunk 3]
```diff
@@ -1,76 +1,88 @@
...
+	public void testBackgroundDisplayedForIntegerParameter() throws Exception {
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/properties/creators/NumberParameterPropertyUICreatorIT.java]: worth adding the other test checking that the background is red when the value is not right

**Comment by bfitzpat** [comment]
I'm attempting to fix it locally  but not having much luck.

**Comment by bfitzpat** [comment]
Updated to require-bundle and second test for invalid value that should show red background.

**Comment by bfitzpat** [comment]
Sounds good. Maybe spend some time on it Monday? Thanks Aurelien

**Comment by apupier** [comment]
yep

**Comment by bfitzpat** [comment]
Apparently we'll have to get another +1 on Monday for this after an update to the commit? Weird.

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/creators/modifylisteners/number/AbstractNumberModifyListener.java [hunk 4]
```diff
@@ -1,51 +1,53 @@
...
+		this.camelModelElement = camelModelElement;
+		this.parameterName = parameterName;
+	}
+
+	@Override
+	public void modifyText(ModifyEvent e) {
+	    Text txt = (Text)e.getSource();
+	    String val = txt.getText();
+	    try {
+	    	if (val != null && val.toString().trim().length() > 0) {
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/creators/modifylisteners/number/AbstractNumberModifyListener.java]: val is already a String, no need to call to String on it

**Pull request merged**